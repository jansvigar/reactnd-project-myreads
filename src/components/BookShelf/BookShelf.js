import React from 'react'
import PropTypes from 'prop-types'
import BooksGrid from '../BooksGrid/BooksGrid'

const BookShelf = props => {
  const updateShelvedBooksList = () => {
    props.updateShelvedBooksList
      && props.updateShelvedBooksList();
  };

  return (
    <div className="bookshelf">
      <h2 className="bookshelf-title">{ props.title }</h2>
      <div className="bookshelf-books">
        <BooksGrid
          books={ props.books }
          updateShelvedBooksList={ updateShelvedBooksList }
        />
      </div>
    </div>
  );
}

BookShelf.propTypes = {
  title: PropTypes.string.isRequired,
  books: PropTypes.array.isRequired
}

export default BookShelf
