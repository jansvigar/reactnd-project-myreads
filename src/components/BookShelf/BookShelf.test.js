import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BookShelf from './BookShelf';

const props = {
  title: 'Book Shelf Title',
  books: [{id:'id', shelf:'testShelf'}]
}

describe('<BookShelf />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<BookShelf {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
