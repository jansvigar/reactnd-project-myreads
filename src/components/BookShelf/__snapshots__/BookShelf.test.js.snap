// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookShelf /> should render correctly 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Book Shelf Title
  </h2>
  <div
    className="bookshelf-books"
  >
    <BooksGrid
      books={
        Array [
          Object {
            "id": "id",
            "shelf": "testShelf",
          },
        ]
      }
      updateShelvedBooksList={[Function]}
    />
  </div>
</div>
`;
